swarm/network/kademlia.go

```diff
@@ -283,6 +283,22 @@ func (k *Kademlia) SuggestPeer() (suggestedPeer *BzzAddr, saturationDepth int, c
 func (k *Kademlia) On(p *Peer) (uint8, bool) {
 	k.lock.Lock()
 	defer k.lock.Unlock()
+
```

Comment:
this wont work for pss only peers in future.
i recommend putting a check in here https://github.com/ethereum/go-ethereum/blob/master/swarm/network/hive.go#L158
if no pss or stream protocol running, do not call On and skip to return on line https://github.com/ethereum/go-ethereum/blob/master/swarm/network/hive.go#L171